src/app/components/my-stats/my-stats.component.ts

```diff
@@ -0,0 +1,115 @@
+                    if(this.pCategories[i].pk === 64){
+                        this.basicsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk ).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    } else if(this.pCategories[i].pk === 72)
+                        this.predefNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 83)
+                        this.conditionalsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 91)
+                        this.loopsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 100)
+                        this.methodsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 107)
+                        this.arraysNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 115)
+                        this.oopNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                }
+
+                this.easies = this.stats.category_stats.filter(stats => stats.difficulty === 'EASY').reduce((sum, obj) => {
+                    return sum + obj.questions_solved
+                },0)
+
+                this.mediums = this.stats.category_stats.filter(stats => stats.difficulty === 'MEDIUM').reduce((sum, obj) => {
+                    return sum + obj.questions_solved
+                },0)
+
+                this.hards = this.stats.category_stats.filter(stats => stats.difficulty === 'HARD').reduce((sum, obj) => {
+                    return sum + obj.questions_solved
+                },0)
```

Comment:
A similar approach should be implemented for the difficulties. You can get the difficulties using `DifficultyService`.